main.go

```diff
@@ -0,0 +1,229 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/charmbracelet/glamour"
+)
+
+// models
+var geminiModels = []string{
+	"gemini-2.0-flash-thinking-exp-01-21",
+}
+
+func main() {
+	// non interactive execution when query is specified on the command line.
+	queryFromCmd := flag.String("query", "", "query for the agent")
+	maxIterations := flag.Int("max-iterations", 20, "maximum number of iterations")
+	kubeconfig := flag.String("kubeconfig", "/Users/sunilarora/.kube/auto-ai-1", "path to the kubeconfig file")
+	llmProvider := flag.String("llm-provider", "gemini", "language model provider")
+	model := flag.String("model", geminiModels[0], "language model")
+	templateFile := flag.String("template", "", "path to custom prompt template file")
+	tracePath := flag.String("trace", "trace.log", "path to the trace file")
+	promptFilePath := flag.String("prompt-file", "prompt.log", "path to the prompt file")
+	removeWorkDir := flag.Bool("remove-workdir", false, "remove the temporary working directory after execution")
+	flag.Parse()
+
+	logFile, err := os.OpenFile("app.log", os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
+	if err != nil {
+		slog.Error("Error opening log file", "error", err)
+		return
+	}
+	defer logFile.Close()
+
+	textHandler := slog.NewTextHandler(logFile, nil)
+	logger := slog.New(textHandler)
+
+	mdRenderer, err := glamour.NewTermRenderer(
+		glamour.WithAutoStyle(),
+		glamour.WithPreservedNewLines(),
+		// glamour.WithWordWrap(2000),
+		glamour.WithEmoji(),
+	)
+	if err != nil {
+		logger.Error("Error initializing the markdown renderer", "error", err)
+		return
+	}
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+
+	logger.Info("Application started", "pid", os.Getpid())
+
+	go func() {
+		sig := <-sigCh
+		logger.Info("Received signal, shutting down...", "signal", sig)
+		logger.Info("Application exiting")
+		logFile.Close()
+		os.Exit(0)
+	}()
+
+	ctx := context.Background()
+	ctx = withLogger(ctx, logger)
+
+	var contentGenerator LLM
+
+	switch *llmProvider {
+	case "gemini":
+		apiKey := getAPIKey(logger)
+		if apiKey == "" {
+			fmt.Println("GEMINI_API_KEY environment variable not set")
+			return // Exit if API key is not set
+		}
+		client, err := initGeminiClient(ctx, logger, apiKey)
+		if err != nil {
+			return // Exit if client initialization fails
+		}
+		defer client.Close()
+
+		contentGenerator = &GeminiLLM{
+			Client: client,
+			Model:  *model,
+		}
+	case "ollama":
+		// plug in ollama
```

Comment:
Should we remove this case for now given that it doesn't do anything?

Reply:
Done.